clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -659,25 +623,25 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
                                         keyByteArray == null ? ConsumerRecord.NULL_SIZE : keyByteArray.length,
                                         valueByteArray == null ? ConsumerRecord.NULL_SIZE : valueByteArray.length,
                                         key, value);
-        } catch (KafkaException e) {
+        } catch (InvalidRecordException e) {
             throw e;
```

Comment:
Which actually raises the question: why `InvalidRecordException` isn't a `KafkaException`? We could introduce `KafkaException` in the hierarchy and remain backwards compatible.